### PR TITLE
[siemensrds] Fix setting of target temperature

### LINKED ISSUE #17695: [siemensrds] Setting target temperature doesn't work anymore
## Expected Behavior

Setting target temperature should be possible using the binding.

## Current Behavior

Target temperature can't be set. Until the end of last winter this was working well.
Now I get this error:
`2024-11-03 16:58:28.600 [WARN ] [ng.siemensrds.internal.RdsDataPoints] - runtime exception in setValue(), type=java.io.IOException, message="Bad Request"`

If I set the target temperature using the mobile app, the binding can get the new value.

FYI @andrewfg 

## Your Environment

* Version used: openHAB 4.2.2 Release Build
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...): Ubuntu Server 24.04

### PR DESCRIPTION
**Pull request by andrewfg**
Resolves #17695 

Signed-off-by: AndrewFG <software@whitebear.ch>

### CODE REVIEW

**Comment by andrewfg** [comment]
> Can I support you somehow fixing this issue?

@oxigen-dev I fixed your issue; you can download the Jar for testing [org.openhab.binding.siemensrds-4.3.0-SNAPSHOT.jar.zip](https://github.com/user-attachments/files/17618642/org.openhab.binding.siemensrds-4.3.0-SNAPSHOT.jar.zip)

File: bundles/org.openhab.binding.siemensrds/src/main/java/org/openhab/binding/siemensrds/internal/RdsHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.siemensrds/src/main/java/org/openhab/binding/siemensrds/internal/RdsHandler.java:324]: Very minor, consider naming to be consistent with line 317:
```suggestion
                            } else if (command instanceof DecimalType decimalCommand) {
                                targetTemperature = decimalCommand.doubleValue();
```

**Review by jlaur — APPROVED** [review]
Thanks! One very minor comment to consider. Should we await user tests before merging?

**Comment by andrewfg** [comment]
>  Should we await user tests before merging?

@jlaur not really necessary as I have tested on my own system. I posted the Jar for the OP to use if he doesn't want to wait for the new release or milestone.

**Comment by andrewfg** [comment]
> we can also backport to 4.2.x?

Sure.

**Comment by oxigen-dev** [comment]
> > Can I support you somehow fixing this issue?
> 
> @oxigen-dev I fixed your issue; you can download the Jar for testing [org.openhab.binding.siemensrds-4.3.0-SNAPSHOT.jar.zip](https://github.com/user-attachments/files/17618642/org.openhab.binding.siemensrds-4.3.0-SNAPSHOT.jar.zip)

Thanks for the very quick support, I can confirm it is working again!

**Pull request merged**
